thomaspoignant/go-feature-flag
# fix: file exporter path handling

## Description

- Problem: 
  - `OutputDir` in file exporter does not accept non-existent path.
  - If OutputDir is an empty string, the file path will become /{file-name}.json, which would cause it to be treated as an absolute path (instead of a relative path to the current working directory of the process).
-> This will result in a `permission denied` error.
  - The documentation recommends that `OutputDir` should end with `/`.
  However, the current implementation makes the path look like `{OutputDir}//{file-name}.json`, resulting in a redundant `/`.

- The way it was resolved:
Check the validity of OutputDir and create if necessary

- How to test the change:
Configure OutputDir with cases: path does not exist, path with trailing /, path is empty string, and observe the results

## Closes issue(s)

Resolve #2483 

## Checklist
